cdap-common/src/main/java/co/cask/cdap/common/service/RetryOnStartFailureService.java

```diff
@@ -114,7 +110,14 @@ protected void doStop() {
     // the setting of the startedService field. When that happens, the stop failure state is not propagated.
     // Nevertheless, there won't be any service left behind without stopping.
     if (startedService != null) {
```

Comment:
@zuxqoj 
refer here for .oncallback(
it is one of the scenarios where we are having problem